c/meterpreter/source/extensions/stdapi/server/sys/config/config.c

```diff
@@ -590,14 +590,93 @@ DWORD add_windows_os_version(Packet** packet)
 		{
 			if (v.dwMinorVersion == 0)
 			{
-				if (v.dwBuildNumber < 17763) {
-					osName = v.wProductType == VER_NT_WORKSTATION ? "Windows 10" : "Windows Server 2016";
-				} else if (v.dwBuildNumber < 20348) {
-					osName = v.wProductType == VER_NT_WORKSTATION ? "Windows 10" : "Windows Server 2019";
-				} else if (v.dwBuildNumber < 22000) {
-					osName = v.wProductType == VER_NT_WORKSTATION ? "Windows 10" : "Windows Server 2022";
-				} else {
-					osName = v.wProductType == VER_NT_WORKSTATION ? "Windows 11" : "Windows Server 2022";
+				if (v.wProductType == VER_NT_WORKSTATION) {
+					if (v.dwBuildNumber < 10586) {
+						osName = "Windows 10";
+					}
+					else if (v.dwBuildNumber < 14393) {
+						osName = "Windows 10_1511";

```

Comment:
The "1511" is the version here in this context. I've never seen it appended to "Windows 10" with an underscore though, IIRC I've always seen it appended with a space.
```suggestion
						osName = "Windows 10 1511";

```

This update should be applied to all of them. If there's a convention that uses the `_` somewhere that I'm not aware of, just let me know.